backend/src/types.ts

```diff
@@ -267,19 +267,30 @@ export interface INEUPrereqCourse {
  * @param scheduled The scheduled courses.
  */
 export interface ISchedule {
-    completed: ICompleteCourse[];
-    scheduled: string[][];
-    // todo: scheduled: IScheduleCourse[][];
+  completed: IdMap<ICompleteCourse>;
+  scheduled: IdMap<IScheduleCourse>;
+  semesters: IdMap<ISemester>;
+}
+
+/**
+ * A generic type for a map of an id to an object
+ */
+export interface IdMap<T> {
+  [id: string]: T;
 }
 
 /**
  * A scheduled course.
  * @param classId The course number of the scheduled course.
  * @param subject The subject of the scheduled course.
+ * @param title The name of the course, ex "Logic and Computation"
+ * @param dndId A unique id for the course for drag and drop purposes, ex "class-1"
  */
 export interface IScheduleCourse {
-    classId: number;
-    subject: string;
+  classId: number;
+  subject: string;
+  title: string;
+  dndId: string;
```

Comment:
I'm not sure what this is, but it may exist on the more detailed interface for a course that we currently use as part of a schedule, `ScheduleCourse`.